homeassistant/components/rflink/binary_sensor.py

```diff
@@ -81,6 +83,12 @@ def __init__(
         self._delay_listener = None
         super().__init__(device_id, **kwargs)
 
+    async def async_added_to_hass(self):
+        """Restore RFLink BinarySensor state."""
+        await super().async_added_to_hass()
+        if (old_state := await self.async_get_last_state()) is not None:
+            self._state = old_state.state == STATE_ON
+
```

Comment:
You are forgetting to trigger the off_delay here. So the entity will stay on forever if it was on when HA shut down last time.

Reply:
Ps.. in rfxtrx we just directly GO to off if there is a off delay set. That delay likely has timed out already due to the restart.

Reply:
Fixed for devices with `off_delay` setting.
Thank you very much for the tip.